tools/sense_studio/static/testing.js

```diff
@@ -0,0 +1,120 @@
+
+function addTerminalMessage(message) {
+    let terminal = document.getElementById('testTerminal');
+    terminal.insertAdjacentHTML('beforeend', `<div class='monospace-font'><b>${message}</b></div>`);
+    terminal.scrollTop = terminal.scrollHeight;
+}
+
+function streamVideo(message) {
+    let frame = document.getElementById('frame');
+    frame.src = message.image;
+}
+
+async function getPathInputs(pathLabelId, pathInputId) {
+    let pathLabel = document.getElementById(pathLabelId);
+    let pathInput = document.getElementById(pathInputId);
+    let project = document.getElementById('project');
+
+    let name = project.value;
+    let path = pathInput.value;
+
+    let directoriesResponse = await browseDirectory(path, name);
+    let disabled = false;
+
+    // Check that project path is filled and exists
+    if (path === '') {
+        setFormWarning(pathLabel, pathInput, '');
+        disabled = true;
+    } else if (!directoriesResponse.path_exists) {
+        setFormWarning(pathLabel, pathInput, 'This path does not exist');
+        disabled = true;
+    } else {
+        setFormWarning(pathLabel, pathInput, '');
+    }
+}
+
+async function startTesting(url){
+    let classifier = document.getElementById('classifier').value;
+    let inputVideoPath = document.getElementById('inputVideoPath').value;
+    let outputVideoName = document.getElementById('outputVideoName').value;
```

Comment:
What happens if I enable "Save Output Video", enter a name in the field, and then disable it again? I think the video will still be saved, because we don't check the flag again here.

Reply:
What if on toggle back we clear the input field? This case will be same for the input source radio button.